fuse_examples/classification/knight/baseline/clinical_processor.py

```diff
@@ -44,16 +43,61 @@ def __init__(self, json_filename: str, columns_to_tensor: Optional[Union[List[st
                         When None (default) no columns are converted.
         """
         self.json_filename = json_filename
-        self.data = create_knight_clinical(json_filename)
+        self.data = pd.read_json(json_filename)
         self.columns_to_tensor = columns_to_tensor
-        self.sample_desc_column = "SubjectId"
+        self.sample_desc_column = 'case_id'
+        self.selected_column_names = ['case_id', 'age_at_nephrectomy', 'body_mass_index', 'gender', 'comorbidities', \
+                                      'smoking_history', 'radiographic_size', 'last_preop_egfr', 'aua_risk_group']
 
     def __call__(self, sample_desc: Hashable):
         """
         See base class
         """
         # locate the required item
         items = self.data.loc[self.data[self.sample_desc_column] == str(sample_desc)]
+        # keep only selected columns
+        items = items[self.selected_column_names]
+        # process fields
+        if items.gender.iloc[0] == 'male':
+            items.gender.iloc[0] = 0
+        else:
+            items.gender.iloc[0] = 1
+        comorbidities = 0
+        for key, value in items.comorbidities.iloc[0].items():
+            if value:
+                comorbidities = 1
+        items.comorbidities.iloc[0] = comorbidities
+        if items.smoking_history.iloc[0] == 'never_smoked': 
+            items.smoking_history.iloc[0] = 0
+        elif items.smoking_history.iloc[0] == 'previous_smoker':
+            items.smoking_history.iloc[0] = 1
+        elif items.smoking_history.iloc[0] == 'current_smoker':
+            items.smoking_history.iloc[0] = 2
+
+        if items.last_preop_egfr.iloc[0]['value'] == '>=90':
+            items.last_preop_egfr.iloc[0] = 90
+        else:
+            items.last_preop_egfr.iloc[0] = items.last_preop_egfr.iloc[0]['value']
+
+        # Task 1 labels:
```

Comment:
This should go the GT processor only, right?
I guess that already have it in your GT processor

Reply:
I created now a separate GT processor, so now each processor only uses its corresponding data. please review. earlier both GT and input data used the same processor